tests/test_browser_model.py

```diff
@@ -11,17 +10,25 @@ class TestBrowserModel(object):
     def test_new_private_window(self, selenium):
         """Tests opening a new private browsing window via menu"""
         foxpuppet = FoxPuppet(selenium)
-        foxpuppet.browser.open_window(private=True)
-        assert len(foxpuppet.windows.all) == 2
-        assert foxpuppet.browser.is_private is False
-        foxpuppet.windows.focus(foxpuppet.windows.all[1])
-        assert foxpuppet.browser.is_private is True
+        wm = foxpuppet.window_manager
+        wm.windows[0].handle = selenium.current_window_handle
+        print(wm.windows[0].handle)
```

Comment:
This code is totally confusing. Why are you setting the handle of a window outside of the class?